lib/app/worker.rb

```diff
@@ -38,6 +40,19 @@ def pre_flight_check
         )
       end
 
+      def start_heartbeat_task
+        interval_seconds = 60 # seconds
+        Utility::Logger.debug("Starting heartbeat timer task with interval #{interval_seconds} seconds.")
+        task = Concurrent::TimerTask.new(execution_interval: interval_seconds) do
+          Utility::Logger.debug("Sending heartbeat for the connector")
+          Core::ElasticConnectorActions.heartbeat(App::Config['connector_package_id'])
+        rescue StandardError => e
+          Utility::ExceptionTracking.log_exception(e, 'Heartbeat timer terminated due to unexpected error.')
```

Comment:
From reading in [the TimerTask rubydocs](https://ruby-concurrency.github.io/concurrent-ruby/1.1.4/Concurrent/TimerTask.html), I don't think catching this exception will actually terminate the task - it'll just log this message out, then run the task again in `interval_seconds`. I see three approaches:
- you could  pass the task as a param to the block like 
```
do |task|
```
and then shut it down in the rescue
```
rescue StandardError => e
  log_exception("message")
  task.shutdown
```  

- or you could attach an `Observer` that watches for errors and terminates the heartbeat task if it sees one.
- or you could just log the exception and move on, not trying to shut down the heartbeat

Reply:
> it'll just log this message out, then run the task again in `interval_seconds`.
> 
> * or you could just log the exception and move on, not trying to shut down the heartbeat

I think this is expected behaviour?